### DISCUSSION TITLE
JUCE

### DISCUSSION THREAD

**Discussion by briwil** [FundamentalFrequency/.github#64] [Software]
I've been trying to get aquatinted with Juce recently in hopes of being able to add features to the current firmware, but I see the available code has already been exported from Juce; is there a reasonable way to bring the code back into a Juce project? Or is my understanding of Juce still too limited, maybe there is no need to do so and I should just open the code in Xcode?

**Comment by stonepreston** [comment]
LMN uses CMake as a build system. It is a JUCE project already, it just does not use the projucer (CMake is used as an alternative to the projucer, a superior alternative imo). If you install CMake and llvm (you need clang and lld from llvm) you should be able to build from the command line using the commands outlined in the DAW repo.  

If you want xcode integration (ie click the build button in XCode and it builds the project), it looks like you can use the -GXcode flag when configuring:
```
cmake -B build -G Xcode -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS=-fuse-ld=lld
```
That will generate an xcode project you can open up. 

But thats not really necessary, you can still just open the code up in Xcode or any other editor and edit what you want then build at the command line.
